src/clientiface.cpp

```diff
@@ -90,13 +91,26 @@ void RemoteClient::GetNextBlocks (
 		return;
 	}
 
+	// get view range and camera fov from the client
+	s16 wanted_range = sao->getWantedRange();
+	float camera_fov = sao->getFov();
+
+	const s16 full_d_max = std::min(g_settings->getS16("max_block_send_distance"), wanted_range);
```

Comment:
g_settings->getS16("max_block_send_distance") and the other variable should be cached somewhere

Reply:
This is part of the code that loads blocks, and sends them over the network, i.e. not hot. This seems overkill.

Reply:
Also surely it being 'const' means it is as good as cached?

Reply:
const means only we cannot modified it, its for dev, not for runtime

Reply:
How about 'static const'?

Reply:
Ok yes i misunderstood 'const'. So i agree with nerzhul some optimisation is needed here.

Reply:
I do not agree at all. This is not a hot code path and this is not needed.

(It's not even clear where you'd cache it. Can't do a static since this value can be changed without ending the executable. In any other case, how often should we refresh it?)

Reply:
> Can't do a static since this value can be changed without ending the executable

It could be, but we actually have many settings that are used in a static const for performance reasons, so that the user has to restart the program for a change to take effect. These 2 settings are ones that are not changed often by a user so it would be ok to use static const for these.

However these are not high priority for optimisation so i personally am not insistiing on it.

Reply:
Sorry indeed these can't be static const since there is a min with wanted range.

Reply:
Not sure if this is possible but i would cache these as private members of `class RemoteClient`, initialise them in the constructor?

Reply:
However it's not high priority so i might still +1 if you do not optimise, but i would prefer you do if it is possible.

Reply:
I really do not think that is necessary. There are many more configs that we retrieve this way - not an excuse not to fix it, but in this code it is simply not needed and will not help with code readability.

We could have a general PR to think about caching of configs.

Reply:
Ok i won't insist on it.